### PR TITLE
Added program to count set bits

### PR DESCRIPTION
**Pull request by VipulRaj-123**
#### Description of Change
This is a bit manipulation program which outputs the number of set bits(1's) in binary representation of a number.  

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1509"></a>

### CODE REVIEW

**Comment by VipulRaj-123** [comment]
Actually I'm a newbie to open source that is why the 3 commits leaving the latest one had some issue which I had to remove after carefully observing the code documentation.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Code is not up to the [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md).
Please read them **carefully** and **follow them**. Thanks.

File: bit_manipulation/Count_of_set_bits.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/** * @file
+ * @brief program to Count set bits in an integer 
+ * @details
+ * Time Complexity :- O(log n)
+ * Space complexity :- O(1)
+
+**/
+#include <iostream>
+using namespace std;
```
Comment (manncodes) [inline bit_manipulation/Count_of_set_bits.cpp:9]: using namespace std is considered bad practice.
Please make scope specific namespaces. [Reference](https://www.geeksforgeeks.org/using-namespace-std-considered-bad-practice/)

File: bit_manipulation/Count_of_set_bits.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    return count;
+}
+ 
+/* Main Program to test function countSetBits */
+int main()
+{
+    cout << countSetBits(19)<<endl; //binary representation is 10011 so set bits=3
+    cout << countSetBits(23)<<endl; //binary representation is 10111 so set bits=4
+    cout << countSetBits(25)<<endl; //binary representation is 11001 so set bits=3
```
Comment (manncodes) [inline bit_manipulation/Count_of_set_bits.cpp:28]: Make a seperate function `self_test()` for implementing a self test. If you make a stress test, kudos to you!

File: bit_manipulation/Count_of_set_bits.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+**/
+#include <iostream>
+using namespace std;
+ 
+/* Function to get no of set bits in binary
+representation of positive integer n */
+unsigned int countSetBits(unsigned int n)
+{
+    unsigned int count = 0; // "count" variable is used to count number of 1's in binary representation of the number
```
Comment (manncodes) [inline bit_manipulation/Count_of_set_bits.cpp:15]: Consider using uint64_t for non-negative values (or their appropriate size: uint32_t, uint16_t, uint8_t) or int64_t for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Comment (VipulRaj-123) [reply bit_manipulation/Count_of_set_bits.cpp:15]: I will surely do sir

Comment (manncodes) [reply bit_manipulation/Count_of_set_bits.cpp:15]: And easy way to achieve repository standards is to open this in gitpod, run `clang-tidy` & `clang-format` as given [here](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Further you can enable github-action to your fork for automating `clang-tidy` & `clang-format` after every push!

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
@VipulRaj-123, please check the [contributing guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md) for your code to be up to our standards and get merged.

**Pull request closed without merging**